hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/adapter/Spark3_3Adapter.scala

```diff
@@ -124,8 +124,4 @@ class Spark3_3Adapter extends BaseSpark3Adapter {
     case OFF_HEAP => "OFF_HEAP"
     case _ => throw new IllegalArgumentException(s"Invalid StorageLevel: $level")
   }
-
-  override def failAnalysisForMIT(a: Attribute, cols: String): Unit = {
```

Comment:
same here for `HoodieSpark33CatalystPlanUtils`

Reply:
Moving to `HoodieSpark33CatalystPlanUtils` is producing compile error:

```
%  mvn clean -T 12C package -DskipTests -Dspark3.3 -Dscala-2.12
[ERROR] /Users/amrish/code/amrish-hudi-1/hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/HoodieSpark33CatalystPlanUtils.scala:62: error: value failAnalysis is not a member of org.apache.spark.sql.catalyst.expressions.Attribute
[ERROR]     a.failAnalysis(s"cannot resolve ${a.sql} in MERGE command given columns [$cols]")
[ERROR]     
```

Reply:
Fixed. Add failAnalysisForMIT function to HoodieSpark[32,33]CatalystPlanUtils.scala